File: web/src/api/backend/system/field.ts
/**
 * 字段权限 API
 * @module api/backend/system/field
 */
import { adminRequest } from '@/utils/http'

/**
 * 获取所有资源列表（用于菜单resource字段选择）
 */
export function fetchResourceList() {
  return adminRequest.get<{ list: Array<{ code: string; label: string }> }>({
    url: '/field/resourceList'
  })
}

File: server/internal/controller/admin/field.go
package admin

import (
	"context"

	api "xygo/api/admin"
	"xygo/internal/field"
	"xygo/internal/model/input/adminin"
)

// ResourceList 获取所有已注册的资源列表
func (c *ControllerV1) ResourceList(ctx context.Context, req *api.ResourceListReq) (res *api.ResourceListRes, err error) {
	resources := field.GetAll()

	items := make([]adminin.ResourceItem, 0, len(resources))
	for _, r := range resources {
		items = append(items, adminin.ResourceItem{
			Code:  r.Code,
			Label: r.Label,
		})
	}

	res = new(api.ResourceListRes)
	res.ResourceListModel = &adminin.ResourceListModel{
		List: items,
	}
	return res, nil
}
